### ISSUE TITLE
Salt pkg.installed false error.

### ISSUE THREAD

**Issue by AkhterAli** [saltstack/salt#41819]
### Description of Issue/Question

Salt pkg.installed (on rhel based machine) thinks packages that are already installed is an error.
This **only** occurs in **2016.11.5** and **not** **2016.11.2**

### Setup
```
yum_deps:
  pkg.installed:
    - pkgs:
      - python-setuptools
      - python-pip
```

### Steps to Reproduce Issue
Run the state above twice and you are returned with
```
[INFO    ] Executing command ['yum', '-y', 'install', 'python-pip', 'python-setuptools'] in directory '/root'
[TRACE   ] stdout: Loaded plugins: priorities, update-motd, upgrade-helper
Package python26-pip-6.1.1-1.23.amzn1.noarch already installed and latest version
Package python26-setuptools-12.2-1.32.amzn1.noarch already installed and latest version
Nothing to do
[ERROR   ] The following packages failed to install/update: python-pip, python-setuptools
```

### Versions Report
```
Salt Version:
           Salt: 2016.11.5

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.1
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Sep  1 2016, 22:14:00)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.1
            ZMQ: 4.1.6

System Versions:
           dist:
        machine: x86_64
        release: 4.4.35-33.55.amzn1.x86_64
         system: Linux
        version: Not Installed
```

**Comment by terminalmage** [comment]
As you can see in the log, the actual package names are `python26-pip` and `python26-setuptools`. Salt does not support using aliased package names in the `pkg.installed` state. The reason for this is that the state has to do the following:

1. Check if targeted packages are installed
2. For those that are not, perform the install
3. Check again to see if targeted packages are installed

It is using the names you provided to do the checks in step 1 and 3 above, which is why you are getting that error.

Updating your SLS file to use `python26-setuptools` and `python26-pip` will allow the state to pass.

**Comment by AkhterAli** [comment]
While I understand the logic behind the failure, I don't understand why it isn't based off of an exit code or a more complex logic that checks yum installed packages.

**Comment by terminalmage** [comment]
Steps 1 and 3 in my previous comment already check the installed packages, that's how they know whether or not the package you told Salt to install is actually installed.

For what it's worth, we did at one time have a mechanism to check for what packages provide a given package, if the package name specified was not itself available, using a yum/dnf plugin. The problem was that it was painfully slow, adding in some cases 10 seconds or more to a single ``pkg.installed`` state. The choice was made to ditch that code for yum/dnf because performance was more important than the occasional benefit from being able to report to the user that they're using the wrong package name.

Also, I don't believe that this only occurs in 2016.11.5 and not 2016.11.2. We removed this "provides" support a long time ago, certainly well before 2016.11.0 was ever released. This appears to be happening because Amazon has created ``python26-pip`` with a ``Provides: python-pip`` in its spec file. In the case where you were using 2016.11.2, I suspect you're using a different platform, or failing that a different AMI where the repository doesn't have the same ``Provides`` metadata in their pip RPM.
